src/Microsoft.Extensions.NamedDependencyInjection/ServiceProviderExtensions.cs

```diff
@@ -64,5 +64,58 @@ public static object GetService<TKey>(this IServiceProvider serviceProvider, Typ
+        /// <typeparam name="TKey">The type of key.</typeparam>
+        /// <param name="serviceProvider">The System.IServiceProvider to retrieve the service object from.</param>
+        /// <param name="key">A key on which the dependency is registered.</param>
+        /// <returns>A service object of type T that registered by specified key.
+        /// If there are no service or key - exception will be throwed.</returns>
+        /// <exception cref="InvalidOperationException">There is no service of type serviceType that registered by specified key.</exception>
+        public static T GetRequiredService<T, TKey>(this IServiceProvider serviceProvider, TKey key)
+        {
+            var service = serviceProvider.GetRequiredService(typeof(T), key);
+            return (T)service;
+        }
+
+        /// <summary>
+        /// Gets the service object of the specified type by specified key of type TKey.
+        /// </summary>
+        /// <typeparam name="TKey">The type of key.</typeparam>
+        /// <param name="serviceProvider">The System.IServiceProvider to retrieve the service object from.</param>
+        /// <param name="serviceType">An object that specifies the type of service object to get.</param>
+        /// <param name="key">A key on which the dependency is registered.</param>
+        /// <returns>A service object of type serviceType that registered by specified key.
+        /// If there are no service object or key - exception will be throwed.</returns>
+        /// <exception cref="InvalidOperationException">There is no service of type serviceType that registered by specified key.</exception>
+        public static object GetRequiredService<TKey>(this IServiceProvider serviceProvider, Type serviceType, TKey key)
+        {
+            var service = serviceProvider.GetService(serviceType, key)
+                ?? throw new InvalidOperationException($"No service for type '{serviceType.FullName}' with key '{key}' has been registered.");
+            return service;
+        }
+
+        /// <summary>
+        /// Gets the service of type T by specified key comparer.
+        /// </summary>
+        /// <typeparam name="T">The type of service object to get.</typeparam>
+        /// <typeparam name="TKey">The type of key.</typeparam>
+        /// <param name="serviceProvider">The System.IServiceProvider to retrieve the service object from.</param>
+        /// <param name="keyComparer">A comparer function.</param>
+        /// <returns>A service object of type T that registered by specified key.
+        /// If there are no service or key - exception will be throwed.</returns>
+        /// <exception cref="InvalidOperationException">There is no service of type serviceType that registered by specified key.</exception>
+        public static T GetRequiredService<T, TKey>(this IServiceProvider serviceProvider, Func<TKey, bool> keyComparer)
+        {
+            var service = serviceProvider.GetService<T, TKey>(keyComparer);
```

Comment:
can I ask you also to change this expression to ```?? throw new...``` as you've done in another method. Thank you